DTLOR_DP.py

```diff
@@ -107,313 +126,221 @@ def DP(hostTree, parasiteTree, phi, locus_map, D, T, L, Origin, R):
-                        for l2 in valid(l_bottom, allsynteny):
-                            ep2_switch=C[(ep1, eh, l1)] + bestSwitch[(ep2, eh, l2)]
-                            ep1_switch=C[(ep2, eh, l2)] + bestSwitch[(ep1, eh, l1)]
-                            min_switch=min(ep2_switch, ep1_switch)
-                            switch_cost = T + delta(l_bottom,l1, Origin, R) + delta(l_bottom,l2, Origin, R)+ min_switch
-                            #TODO: apply combine
-                            if switch_cost < SWITCHepeh:
-                                SWITCHepeh = switch_cost
-                                # if ep2 switching has the lowest cost
-                                if ep2_switch < ep1_switch:
-                                    for location in bestSwitchLocations[(ep2,vh,l2)]:
-                                        currentLoc = location[1] # Switch landing site
-                                        switchList.append(["T", (ep1, vh, l1), (ep2, \
-                                            currentLoc, l2)])
-                                # if ep1 switching has the lowest cost
-                                elif ep1_switch < ep2_switch: 
-                                    for location in bestSwitchLocations[(ep1,vh, l1)]:
-                                        currentLoc = location[1]
-                                        switchList.append(["T", (ep2, vh, l2), \
-                                            (ep1, currentLoc, l1)])
-                                # if ep1 switching has the same cost as ep2 switching
-                                else: 
-                                    for location in bestSwitchLocations[(ep2, vh, l2)]:
-                                        currentLoc = location[1]
-                                        if currentLoc != None:
-                                            switchList.append(["T", (ep1, vh, l1), (ep2, \
-                                            currentLoc, l2)])
-                                        else:
-                                            switchList.append(["T", (ep1, vh, l1), \
-                                                (ep2, currentLoc, l2)])
-                               
-                                    for location in bestSwitchLocations[(ep1,vh, l1)]:
-                                        currentLoc = location[1]
-                                        if currentLoc != None:
-                                            switchList.append(["T", (ep2, vh, l2), \
-                                            (ep1, currentLoc, l1)])
-                                        else:
-                                            switchList.append(["T", (ep1, vh, l1), \
-                                                (ep2, currentLoc, l2)])           
-                    if switchList==[]:
-                        switchList=[[Infinity]]
-
+                    def get_transfer(l1, l2):
```

Comment:
I think we only need one location here since the transferred child always has the same location as its parent? We can also discuss this more with Prof. Bush to see if this statement is actually well-supported by biological evidence.